### ISSUE TITLE
Fail in import OpenPose to Unity

### ISSUE THREAD

**Issue by jImIjIn** [CMU-Perceptual-Computing-Lab/openpose#201]
### Issue summary
I'm trying to use OpenPose in Unity. I compile the "extract from image" tutorial into a DLL through VS2015 and import to Unity. The Unity crashes in the Caffe part. If I comment out the Caffe part and make a new DLL, everything works fine.  Thank you very much

### OpenPose output
> Unhandled exception at 0x00007FF9F44E34BE (ucrtbase.dll) in Unity.exe: Fatal program exit requested.

Break in `netCaffe.cpp`

`upCaffeNet.reset(new caffe::Net<float>{mCaffeProto,  caffe::TEST});`

### Type of issue
- Help wanted

### Your system configuration
**Operating system** : Windows10
**CUDA version** : 8.0
**cuDNN version**: 5.1
**GPU model** : GeForce GTX 1080 
**Caffe version**: Default from OpenPose or custom version.
**OpenCV version**: installed with `apt-get install libopencv-dev` (Ubuntu) or default from OpenPose (Windows) or OpenCV 2.X or OpenCV 3.X.

**Comment by jImIjIn** [comment]
I'm just being stupid, please ignore me. I didn't put the model under unity root folder

**Comment by marwie** [comment]
fdg

**Comment by jerryMeng100** [comment]
Interested in this as well, were there any other things along the way?  Do you just convert the entire demo into a DLL and port it?

**Comment by JimiJin** [comment]
Yes, I output position data into unsigned char array and read it in Unity

**Comment by jerryMeng100** [comment]
I made my own exportable function to be put into a dll (need to put in array still but idea is there) but the common language support is turned off in VS2015 and Unity's not seeing the dll.  Think it might be a .NET version error since by default clr is off for the example project, did you run into this?  I turned it on but it gave me "thread is not supported" or similar.

**Comment by jerryMeng100** [comment]
Whoops ^ was not the case at all, dumb mistake.  I'm now converting the 1 extract image demo and getting some dependency issues (module not found when loading dll in Unity, I'm kinda new at dealing with dlls and the sort).  Is there a chance we can go to an email thread?  My email is zmeng90@stanford.edu.  Likely making a lot of stupid mistakes so any help is appreciated!

**Comment by georgecfwebtools** [comment]
Hello jjin1223,

I came to the same spot and would use OpenPose in Unity, I compiled the same "extract from image" tutorial to dll in VS 2015 and tested with C#, it worked outside Unity.

Within Unity, I created a simple game and imported the dll as well as all the other dlls required by OpenPose into plugins folder. Imported the models folder to root folder, import the image.

When I call the dll within Unity, it crashed.

Would it be possible for you to give a little details how you make it work? It seems easy but there is a lot of details like where to put dlls, models, images etc.

Thank you!

George
